frontend/src/components/ListingPanel/ListingPanel.jsx

```diff
@@ -4,68 +4,96 @@ import { getImageFromId } from "../../utils/imageService";
-          <img className="listing-img" src={image} />
-        </div>
-        <div className="listing-l">
-          <span className="listing-date">Posted X Days Ago</span>
-          <h2 className="listing-title">{listing.title}</h2>
-          <h3 className="listing-address">{hasAddress ? listing.location.address : "No Address"}</h3>
-          <div className="listing-details">
-            <h4>Details</h4>
-            <div className="details">
-              <div className="details-l">
-                <span>Quantity</span>
-                <span>Condition</span>
-                <span>Categories</span>
-              </div>
-              <div className="details-r">
-                <span>{listing.details.quantity}</span>
-                <span>{listing.details.condition}</span>
-                <span>{listing.details.categories}</span>
-              </div>
-            </div>
-          </div>
-          <div className="listing-description">
-            <h4>Description</h4>
-            <p className="description-text">{listing.description}</p>
-          </div>
-          <div>
+		if (listing.claim_queue.indexOf(currentUser.uid) === -1) {
+			listing.claim_queue.push(currentUser.uid);
+			listing.claimed = true;
+			await updateListingById(listing._id, listing);
+		}
+	};
 
-          </div>
-          <button className="claim-btn" disabled={listing.claimed} onClick={addUserToClaimQueue}>
-            Claim
-          </button>
-        </div>
-      </div>
-    </div>
-  );
+	return (
+		<div className="listing-wrapper">
+			<div className="listing">
+				<div className="listing-r">
+					<img className="listing-img" src={image} />
+				</div>
+				<div className="listing-l">
+					<span className="listing-date">Posted X Days Ago {author && `by ${author}`}</span>
+					<h2 className="listing-title">{listing.title}</h2>
+					<h3 className="listing-address">
+						{hasAddress ? listing.location.address : "No Address"}
+					</h3>
+					<div className="listing-details">
+						<h4>Details</h4>
+						<div className="details">
+							<div className="details-l">
+								<span>Quantity</span>
+								<span>Condition</span>
+								<span>Categories</span>
+							</div>
+							<div className="details-r">
+								<span>{listing.details.quantity}</span>
+								<span>{listing.details.condition}</span>
+								<span>{listing.details.categories}</span>
+							</div>
+						</div>
+					</div>
+					<div className="listing-description">
+						<h4>Description</h4>
+						<p className="description-text">{listing.description}</p>
+					</div>
+					<div></div>
+					{currentUser.uid !== listing.user_id && (
+						<Button
+							variant="primary"
+							disabled={claimListing}
+							onClick={addUserToClaimQueue}
+						>
+							{claimListing ? "Claimed" : "Claim"}
+						</Button>
+					)}
+					<p>
+						{listing.claim_queue.length}
+						{listing.claim_queue.length === 1 ? " user has " : " users have "}
+						claimed
```

Comment:
LGTM!
Very minor nit: add "this listing" after claimed just so it sounds more complete.
So it will be "X users have claimed this listing"